modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrainBase.h

```diff
@@ -86,6 +86,9 @@ class ComputeThermalExpansionEigenstrainBaseTempl
 
   /// current temperature
   const GenericVariableValue<is_ad> & _temperature_prop;
+
+  /// mean coefficient of thermal expansion (for output verification)
+  GenericMaterialProperty<Real, is_ad> & _alpha_bar_prop;
```

Comment:
For verification purposes, we may not need a generic property. We can use the MetaPhysicL function `raw_value` to make sure we always store the property value itself with the derivative information, which is only useful to build the Jacobian.

Reply:
Does it consume that much memory? Seems better to keep it consistent?

Reply:
If this material is for analysis purposes, I think consistency would be not to include derivative information. If we are not sure of how this material will be used, then, I agree that for consistency it should be generic. 

I don't think it would have a noticeable impact on memory consumption or computation. I just thought that, if it's not needed, why do it.